Update composer to point to parse-community

acinader opened the pull request:
disclaimer: I haven't used composer in years :).

I updated my composer.json to have:

 `"parse/php-sdk": "^1.2.7",`

when i run `composer update --ignore-platform-reqs parse/php-sdk`

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.


  Problem 1
    - The requested package parse/php-sdk ^1.2.7 exists as parse/php-sdk[1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.0.5, 1.0.6, 1.1.0, 1.1.1, 1.1.10, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.1.6, 1.1.7, 1.1.8, 1.1.9, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, dev-master, 1.1.x-dev, dev-consistent-appersand-in-urls] but these are rejected by your constraint.
```
I don't think that this pr will fix the problem, but figure should be fixed anyway?!?

acinader commented:
https://packagist.org/packages/parse/php-sdk

doesn't show the last two tags and maybe this will fix that?  it also doesn't recognize me even though i am linked via github??

maybe this will fix it.  if not, I can try to figure out how to get permissions for me and @montymxb

montymxb commented on CONTRIBUTING.md:
```diff
@@ -11,7 +11,7 @@ into a single commit. Please provide ample explanation in the commit message.
 ------------
 
-Testing the Parse PHP SDK requires having a working Parse Server instance to run against. 
+Testing the Parse PHP SDK requires having a working Parse Server instance to run against.
```
I've noticed a lot of line endings were adjusted throughout, did you end up normalizing them?

montymxb commented on README.md:
```diff
@@ -299,16 +299,16 @@ if(ParsePush::hasStatus($response)) {
 
     }
-    
+
```
Again here with what was 4 spaces I think, seems to have turned it into a tab? I'm guessing your local setup is converting this for you? Just checking.

acinader replied:
its just trailing whitespace.  its easy to introduce trailing whitespace, especially with our fancy editors that auto indent....

acinader commented:
my editor is (thankfully ;) set to strip trailing white spaces (and also ensure that there is a trailing newline at the end of files.

so what you're seeing is my editor cleaning up any file I edit.  In the node lib, we institutionalized this back in December by running lint as part of our unit tests so no new bad formatting can get introduced.

On my past pr's I removed my formatting fixes by hand, but its cumbersome for me, so I just shrugged and committed em this time.  If you want. I can revert the whitespace changes.

acinader commented:
@montymxb let's see if this change gives me permission with packagist.  it should.  if not, then yeah you or I should see how to get the control we need.

montymxb commented:
Sounds good! I guess we'll keep our eyes open to see if this works now...

acinader commented:
@montymxb I went into the settings of this github repo and in the integrations for packagist I put in my packagist username and token -- presto!  1.2.7 is published and i think that all future tags will publish too....

flovilmart commented:
We shall centralize at one point those credentials :)

acinader commented:
@flovilmart it seems like packagist is doing the right thing. 

to allow packagist to receive post commit notifications from github, a committer from the repo needs to put her packagist token into the settings of the github repo to make an authorized link between github and packagist.  

I don't know if what caused it to break was changing the repo path or that the linked user was no longer a committer?  or both?  I suspect it was the change in the repo path cause it still shows fosco (well it did, i just removed fosco and added montymxb)

So there's no real credentials to centralize.  It appears to be mindful of rights to the repo as it should

flovilmart commented:
yeah, I mean if we have such integrations, we may a parse-community bot, so we'd use those credentials instead of personal ones no?

acinader commented:
My point is that there are no ongoing credentials.  to CREATE a link, a token from an authorized member of the repo are needed, but they serve no purpose in an ongoing way.

I'm sure there are other cases where we need a solution for credentials, this just doesn't appear to be one of them :).

flovilmart commented:
ok!

montymxb commented:
@acinader yea it looks like the repo url was simply out of date. Upon creating the packagist repo I believe that was set in stone. It sounds like after I reached out to gfosco that's all he had to change for us.

@flovilmart The service hook from github via your credentials doesn't actually change the behavior of how packagist reads this repo, it just makes it run after a change is made 👍 . Otherwise after pushing a release we would have to wait until packagist.org decides to check this repo again before composer users would see any changes. This way the updates should be near instantaneous :). 

Packagist apparently doesn't want anonymous users polling their repo update API, and they restrict it to maintainers/owners of a project for those that can. Either way the existing functionality remains unchanged, except for the timeliness of it.
